core/iwasm/compilation/aot_compiler.c

```diff
@@ -441,32 +441,49 @@ aot_compile_func(AOTCompContext *comp_ctx, uint32 func_index)
                     return false;
                 break;
 
-#if WASM_ENABLE_REF_TYPES != 0
+#if WASM_ENABLE_REF_TYPES != 0 || WASM_ENABLE_GC != 0
             case WASM_OP_SELECT_T:
             {
                 uint32 vec_len;
 
-                if (!comp_ctx->enable_ref_types) {
-                    goto unsupport_ref_types;
+                if (!comp_ctx->enable_ref_types && !comp_ctx->enable_gc) {
+                    goto unsupport_gc_and_ref_types;
                 }
 
                 read_leb_uint32(frame_ip, frame_ip_end, vec_len);
                 bh_assert(vec_len == 1);
                 (void)vec_len;
 
                 type_idx = *frame_ip++;
-                if (!aot_compile_op_select(comp_ctx, func_ctx,
-                                           (type_idx != VALUE_TYPE_I64)
-                                               && (type_idx != VALUE_TYPE_F64)))
-                    return false;
+#if WASM_ENABLE_GC != 0
+                if (comp_ctx->enable_gc) {
+                    if (!aot_compile_op_select(
+                            comp_ctx, func_ctx,
+                            (type_idx != VALUE_TYPE_I64)
+                                && (type_idx != VALUE_TYPE_F64)
+#if UINTPTR_MAX == UINT64_MAX
+                                && (!wasm_is_type_reftype(type_idx))
```

Comment:
A little strange for the modification, WASM_ENABLE_REF_TYPES also has funcref and externref, but it didn't check whether type_idx is funcref or externref.
Seems no need to modify?

Reply:
The previous select_t doesn't take the reference as an argument.

Reply:
```C
                                && (type_idx != VALUE_TYPE_I64)
                                && (type_idx != VALUE_TYPE_F64)
#if UINTPTR_MAX == UINT64_MAX
#if WASM_ENABLE_GC != 0
                                && !(comp_ctx->enable_gc && wasm_is_type_reftype(type_idx))
#endif
#endfi